common/components/common/tags/TagSelect.jsx

```diff
@@ -0,0 +1,71 @@
+// @flow
+
+import React from 'react'
+import Select from 'react-select'
+import _ from 'lodash'
+
+
+
+type Props = {|
+  elementId: string,
+  category: string,
+  value?: string,
+  onSelection: (string) => void
+|};
+type State = {|
+  tags: Array<TagDefinition>,
+  selected: TagDefinition
+|};
+
+/**
+ * Dropdown selector for tags
+ */
+class TagSelect extends React.PureComponent<Props, State> {
+  constructor(props: Props): void {
+    super(props);
+    this.state = {
+      tags: [],
+      selected: this.props.value || ""
+    };
+    
+    fetch(new Request('/api/tags?category=' + this.props.category))
+      .then(response => response.json())
+      .then(tags =>
+        tags.map(tag => ({
+          value: tag.tag_name,
+          label: tag.display_name
+        }))
+      )
+      .then(options => this.setState({tags: options}));
+  }
+  
+  componentWillReceiveProps(nextProps: Props): void {
+    if(nextProps.value) {
+      this.setState({selected: {value:nextProps.value, label:_.find(this.state.tags, tag => tag.value === nextProps.value).label} || ""});
```

Comment:
I think this is builtin and you don't need lodash here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find

Reply:
Done